syntax = "proto3";

option cc_enable_arenas = true;

package ray.rpc.syncer;

enum RayComponentId {
  RESOURCE_MANAGER = 0;
}

message RaySyncMessage {
  RayComponentId component_id = 1;
  bytes sync_message = 2;
}

message RaySyncMessages {
  repeated RaySyncMessage sync_messages = 1;
}

service RaySyncer {
  rpc StartSync(stream RaySyncMessages) returns (stream RaySyncMessages);
}
